pkg/api/v1alpha1/kepler_internal_types.go

```diff
@@ -36,9 +36,15 @@ type InternalExporterDeploymentSpec struct {
 	Namespace string `json:"namespace"`
 }
 
+type InternalRedfishSpec struct {
+	RedfishSpec `json:",inline"`
+}
+
 type InternalExporterSpec struct {
 	// +kubebuilder:validation:Required
 	Deployment InternalExporterDeploymentSpec `json:"deployment"`
+
+	Redfish *InternalRedfishSpec `json:"redfish,omitempty"`
```

Comment:
We are going to extend spec for configuring the other metrics as well (such as to enable/disable cgroups metrics and so on) - check https://github.com/sustainable-computing-io/kepler-operator/issues/202. 
Thus, I think we should move Redfish to be under `metric.energy` and resource utilization metrics such as cGroups should go under `metric.resource`. 

Refer to terms in Kepler:
```golang
// https://github.com/sustainable-computing-io/kepler/blob/3d0bad753d68f74a0955f6e25d2a9d8e93507d08/pkg/collector/stats/stats.go#L47
type Stats struct {
	ResourceUsage map[string]*types.UInt64StatCollection
	EnergyUsage   map[string]*types.UInt64StatCollection
}
```

Reply:
This seems to be a bigger change to the API. Can we do this api change in a separate PR?

Reply:
Sure. I think we can update that on a different PR. @sthaha What do you think?